src/ledgered/manifest/tests.py

```diff
@@ -0,0 +1,86 @@
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+from .constants import DEFAULT_USE_CASE
+from .types import Jsonable, JsonDict, JsonSet
+from .utils import getLogger
+
+
+class DuplicateDependencyError(ValueError):
+    pass
+
+
+@dataclass
+class TestsDependencyConfig(Jsonable):
+    __test__ = False  # deactivate pytest discovery warning
+
+    url: str
+    ref: str
+    use_case: Optional[str]
+
+    def __init__(self, url: str, ref: str, use_case: Optional[str] = None) -> None:
+        self.url = url
+        self.ref = ref
+        self.use_case = use_case or DEFAULT_USE_CASE
+
+    @property
+    def dir(self):
+        return f"{self.url}-{self.ref}-{self.use_case}"
```

Comment:
In the extension I'm currently using <reponame>-<usecase> (omitting the ref and the full URL :thinking:)

Reply:
Also, we don't know the complete path to the dir with your implem ? 

In the extension the path is `<tests directory as defined in manifest>/.test_dependencies/<dir>`

Reply:
Yes it was finished, I've replaced the `URL` with the `reponame`.
Why omitting the `ref`? We discussed that we were to integrate it. But it does not really matter in the end, your implem works with itself, and if you migrate to `ledgered` it will also work with itself so it will stay correct (you'll just end up with no more used directories).

About the complete path, yes, I'm trying to make it work with the current implementation, but as the parsing is an oriented tree (children does not have access to their parents), it's very difficult to sink down the `pytest_directory` value. I don't have better solution than dirty patches for now. Still working on it.

Reply:
Should be good now